### ISSUE TITLE
arithmetic with ndarray

### ISSUE THREAD

**Issue by crawfordsm** [astropy/ccdproc#326]
Currently arithmetic between an ndarray and CCDData object is not possible, but raises an error when it is attempted.   It does raise a sensible error, but this might be functionality that we do want to enable.

**Comment by mwcraig** [comment]
My main question here would be how to handle units; certainly it ought to be possible at a minimum to do something like `ccd.add(my_numpy_array * u.adu)`. Should we maybe use the `@support_nddata` decorator for this?

**Comment by MSeifert04** [comment]
I've solved it during the `NDArithmetic`-Rafactoring by just converting the operand to another instance of the same class ``operand = self.__class__(operand)``. This would automatically handle units/masks/etc appropriatly while raising errors in case it is not convertible. Which should be fine because I cannot think how such an operation should be defined if the class doesn't know how to interpret it. It's not a possibility with `CCDData` though because you enforce a unit. 

A possible alternative would be to just edit: https://github.com/astropy/ccdproc/blob/master/ccdproc/ccddata.py#L329 to `elif isinstance(other, (numbers.Number, np.ndarray)):`

That would fail for `add`, `subtract` (because of the unit, except the `CCDData` is `dimensionless_unscaled`) but at least work for `multiply` and `divide`.

**Comment by BrunoSanchez** [comment]
Hello, 
I think I have an corner case here, I've found this error while coding some image reduction
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/home/bruno/.virtualenvs/corral/src/corral/corral/run/step.py", line 32, in run
    generator = step.process(obj) or []
  File "/home/bruno/.virtualenvs/corral/src/corral/toritos/toritos/steps.py", line 430, in process
    for apaw in ccdpawlist]
  File "/home/bruno/.virtualenvs/corral/local/lib/python2.7/site-packages/ccdproc/log_meta.py", line 68, in wrapper
    result = func(*args, **kwd)
  File "/home/bruno/.virtualenvs/corral/local/lib/python2.7/site-packages/ccdproc/core.py", line 608, in subtract_dark
    result = ccd.subtract(master_scaled)
  File "/home/bruno/.virtualenvs/corral/local/lib/python2.7/site-packages/ccdproc/ccddata.py", line 444, in subtract
    result = super(CCDData, self).subtract(other)
  File "/home/bruno/.virtualenvs/corral/local/lib/python2.7/site-packages/astropy/nddata/mixins/ndarithmetic.py", line 532, in subtract
    **kwargs)
  File "/home/bruno/.virtualenvs/corral/local/lib/python2.7/site-packages/astropy/nddata/mixins/ndarithmetic.py", line 641, in _prepare_then_do_arithmetic
    result, init_kwds = operand._arithmetic(operation, operand2, **kwargs)
  File "/home/bruno/.virtualenvs/corral/local/lib/python2.7/site-packages/astropy/nddata/mixins/ndarithmetic.py", line 252, in _arithmetic
    result = self._arithmetic_data(operation, operand, **kwds2['data'])
  File "/home/bruno/.virtualenvs/corral/local/lib/python2.7/site-packages/astropy/nddata/mixins/ndarithmetic.py", line 327, in _arithmetic_data
    result = operation(self.data * self.unit,
TypeError: unsupported operand type(s) for *: 'numpy.ndarray' and 'IrreducibleUnit'
1
```
And it seems to be related to this issue #326.
I could not find any errors in my code yet:
```
dark.header['exptime'] = self.dark.exptime
dark_unbias = ccdproc.subtract_bias(dark, bias)

paw_sub = [ccdproc.subtract_dark(apaw,
                      dark_unbias, exposure_time='exptime',
                       exposure_unit=u.second, scale=True)
                       for apaw in ccdpawlist]
```

Any ideas?
Thanks!

**Comment by MSeifert04** [comment]
@BrunoSanchez Could you open a seperate issue for your case? We may need to know what your "dark" and "bias" are and how they were created.

**Comment by mwcraig** [comment]
Will be fixed in 2.0LTS
